pkg/vendir/fetch/hg/hg.go

```diff
@@ -147,39 +208,39 @@ hgauth.password = %s
 		if err != nil {
 			return fmt.Errorf("Writing %s: %s", hgRcPath, err)
 		}
-		env = append(env, "HGRCPATH="+hgRcPath)
+		t.env = append(t.env, "HGRCPATH="+hgRcPath)
+		cacheID += hgRc
```

Comment:
Since we are creating a temp folder to store auth, isn't that going to invalidate the cache every time we run the command and use an SSH Private Key?

Reply:
hgRc doesn't contain the temp folder path, so no. But as I said in the above thread, we only need the URL.